s3/pkg/datastore/driver/factory.go

```diff
@@ -0,0 +1,23 @@
+package driver
+
+import (
+	backendpb "github.com/opensds/multi-cloud/backend/proto"
+	exp "github.com/opensds/multi-cloud/s3/pkg/exception"
+)
+
+type DriverFactory interface {
+	CreateDriver(detail *backendpb.BackendDetail) (StorageDriver, error)
+}
+
+var driverFactoryMgr = make(map[string]DriverFactory)
```

Comment:
why do we need multiple factories ? can you please point me to the design spec for this, thanks.

Reply:
Hi rhsakarpos, this is the design spec, pls have a check, thanks:
https://github.com/opensds/design-specs/blob/master/specs/daito/Multi_cloud_yig_integration.md#integrate-yig-ceph-pool-as-a-storage-backend-into-datastorageengine